pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/forward/ForwardIndexType.java

```diff
@@ -226,6 +234,14 @@ public String getFileExtension(ColumnMetadata columnMetadata) {
     }
   }
 
+  @Override
+  public List<String> getFileExtensions(@Nullable ColumnMetadata columnMetadata) {
+    if (columnMetadata == null) {
+      return EXTENSIONS;
```

Comment:
When will this be the case (ColumnMetadata is null) ? Also since a given column can have a single and a single kind of forward index, returning the list of extensions isn't helpful to the caller ?

Reply:
> When will this be the case (ColumnMetadata is null) ?

In the current code, never. But given its role as an optional filter, it seems ok to support sending a null filter in case we needed in future.

Ideally we shouldn't have to add this parameter and always return all the lists, but that would break compatibility from the previous implementation. Specifically, in a weird case when more than one file (with different extensions) exists for the segment, the older implementation would return the one that matches with the column metadata.

I don't know if that case is possible, but I have found so weird cases I thought that were impossible related to segments and their indexes that I prefer to be cautious here.

> Also since a given column can have a single and a single kind of forward index, returning the list of extensions isn't helpful to the caller ?

But that is not true for other indexes and we need to honor the common interface. Most indexes only expects a single extension and most of them don't even need the columnMetadata parameter.

If the caller knows that ForwardIndexType is being used, they can call `getFileExtension(ColumnMetadata)`.